demos/embeddings/watch-your-step-cora-demo.ipynb

```diff
@@ -0,0 +1,449 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plt.title(\"Graph log likelihood loss vs epoch\")\n",
+    "plt.xlabel(\"Epoch\")\n",
+    "plt.ylabel(\"Graph log likelihood\")\n",
+    "plt.grid()\n",
+    "plt.plot(history.history['loss'])\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualizing Embeddings\n",
+    "\n",
+    "Now we use TSNE to visualize the embeddings."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "embeddings = np.hstack(\n",
+    "    [model.get_layer('WATCH_YOUR_STEP_LEFT_EMBEDDINGS').embeddings.numpy(),\n",
+    "     model.get_layer('WATCH_YOUR_STEP_RIGHT_EMBEDDINGS').kernel.numpy().transpose()]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sklearn\n",
+    "from sklearn.preprocessing import OneHotEncoder\n",
+    "from sklearn.manifold import TSNE\n",
+    "\n",
+    "nodelist = list(Gnx.nodes)\n",
+    "\n",
+    "node_data = pd.read_csv(os.path.join(data_dir, \"cora.content\"), sep='\\t', header=None)\n",
+    "node_data.set_index(0, inplace=True)\n",
+    "\n",
+    "labels = node_data[node_data.columns[-1]].loc[nodelist]\n",
```

Comment:
Why does this need to be `loc`'d? Aren't Gnx's nodes the same as the `node_data`s?

Reply:
I'm not entirely sure why, but the ordering is different in `node_data` compared to the embeddings. So without the loc to reorder `node_data` the accuracy drops to 22%

Reply:
Ah, right. I guess because the graph is created by adding edges (not explicitly adding nodes), so the order of the nodes is probably related to the order of the edges. Using the new StellarGraph class would change this.